### PR TITLE
make sure TOUGH parts are always at the beginning

### PR DESCRIPTION
**Pull request by ESchuderer**

### CODE REVIEW

**Comment by ESchuderer** [comment]
creep

File: creep.Setup.js [hunk 1]
```diff
@@ -163,6 +163,12 @@ let Setup = function(typeName){
                 parts.splice(index, 1);
                 parts.push(HEAL);
             }
+
+            if( parts.includes(TOUGH) ) {
```
Comment (karlthepagan) [inline creep.Setup.js]: This will move a single `TOUGH` part. Isn't this better handled by allowing part sorting? (using the parts comparator)

Comment (ESchuderer) [inline creep.Setup.js]: That's right, I should not try fixing things when tired. Will look into it later this day.

**Comment by cyberblast** [comment]
`parts.sort(Creep.partsComparator);` (https://github.com/ScreepsOCS/screeps.behaviour-action-pattern/pull/604/files#diff-58c6d49d304798758faaa73cc7ae7dc7R158 ) will already ensure TOUGH parts are first. 
Maybe something else fucked up !? 
some bad overrides? tried delete Memory.modules?

**Comment by ESchuderer** [comment]
Very strange indeed, I've only seen one creep have this behaviour. No overrides except parameters and custom tower code. Will see if it happens again.

**Pull request closed without merging**